core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/CsdLocationFilter.java

```diff
@@ -0,0 +1,33 @@
+package com.sequenceiq.cloudbreak.service.upgrade.image;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.model.catalog.Image;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.StackType;
+
+@Component
+public class CsdLocationFilter implements PackageLocationFilter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsdLocationFilter.class);
+
+    @Override
+    public boolean filterImage(Image image, Image currentImage, StackType stackType) {
+        if (StackType.WORKLOAD.equals(stackType)) {
+            if (image == null || image.getPreWarmCsd() == null || image.getPreWarmCsd().isEmpty()) {
+                LOGGER.debug("Image or CSD parcels are not present. Image: {}", image);
+                return false;
+            } else {
+                List<String> csdList = image.getPreWarmCsd();
```

Comment:
This only check the CSDs, what about the pre-warm parcels?

Reply:
I've created a new validator for pre-warm percels. Please check it.